catalog/ui/catalog-ui-search-api/src/main/java/org/codice/ddf/catalog/ui/filter/FlatFilterBuilder.java

```diff
@@ -37,25 +37,59 @@
  */
 public interface FlatFilterBuilder<T> {
 
+  // BUILDER
+
   T getResult();
 
-  FlatFilterBuilder beginBinaryLogicType(String operator);
+  FlatFilterBuilder end();
+
+  // LOGIC
+
+  FlatFilterBuilder not();
+
+  FlatFilterBuilder and();
+
+  FlatFilterBuilder or();
+
+  // BINARY COMPARISON
```

Comment:
After some investigation we could certainly create all the interfaces to support safety, i.e. a `FilterBuilder` version 2.0, and rearrange the design to use it, but I suspect at this point the work would not be leveraged enough to be worth it.

Filter 2.0 support is currently scoped to the UI only. This means any builders we create will be bound to IO between the UI and the backend. Since all our filters passing back and forth on that channel are fully dynamic (we can't hardcode any parts of those filters) the compile-time safety we get would be moot. No one would be using it in a declarative fashion. It would just be a different flavor of the current use case and runtime errors could still occur if the structure is wrong. 

Current high level design: 
```
[Visitable] (multiple implementations mirroring the FlatFilterBuilders)
   -- accepts --> [Transform visitor] (one "default" implementation)
   -- requires --> [FlatFilterBuilder] (multiple implementations)
```

Alternative could be: 
```
[Visitable] (multiple implementations mirroring the FilterBuilders)
   -- accepts --> [Transform visitor] (one "default" implementation)
   -- requires --> [FlatFilterBuilder] (one "default" implementation)
   -- requires --> [FilterBuilder(v2)] (multiple implementations)
```